app/components/app_consent_component.rb

```diff
@@ -55,20 +55,36 @@ def contact_parent_or_guardian_link(consents)
 
   def consents_grouped_by_parent
     @consents_grouped_by_parent ||=
-      @patient_session.consents.group_by { |consent| summary_for(consent) }
+      @patient_session.consents.group_by(&:summary_with_consenter)
   end
 
   private
 
-  def summary_for(consent)
-    summary =
-      if consent.respond_to?(:response_not_provided?) &&
-           consent.response_not_provided?
-        "Contacted #{consent.name}"
-      else
-        "#{consent.human_enum_name(:response).capitalize} by #{consent.name}"
-      end
-    summary += " (#{consent.who_responded})" unless consent.via_self_consent?
-    summary
+  def render_grouped_consents(consents)
+    first_consent = consents.first
+    first_refused_consent = consents.find_all(&:response_refused?).first
+
+    props = {
+      name: first_consent.name,
+      response:
+        consents.map do |consent|
+          { text: consent.summary_with_route, timestamp: consent.recorded_at }
+        end
+    }
+    unless first_consent.via_self_consent?
+      props[:relationship] = first_consent.who_responded
+      props[:contact] = {
+        phone: first_consent.parent_phone,
+        email: first_consent.parent_email
+      }
+    end
```

Comment:
I found this hard to read on the first pass. I think this scans slightly better, what do you think?

```suggestion
    props.merge(
      relationship: first_consent.who_responded
      contact: {
        phone: first_consent.parent_phone,
        email: first_consent.parent_email
      }) unless first_consent.via_self_consent?
```

Reply:
I tried a variation of this in [b7a8b34](https://github.com/nhsuk/manage-childrens-vaccinations/pull/1206/commits/b7a8b348e2c1ecc34071fea5aa42f8b38b19eb67) – the linter didn't like sticking the `unless` at the end and immediately changed it back it but I've adopted the `merge`.

Reply:
Ya thought that might happen. I'll see if it still reads better ... hopefully I've forgotten how it works sufficiently. 😆